### PR TITLE
Timezones: Include IANA timezone canonical name in TimeZoneInfo

### PR DESCRIPTION
**Pull request by dprokop**
Ref: https://www.iana.org/time-zones , https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
This PR also sets the global timezone for tests to UTC.

### CODE REVIEW

File: packages/grafana-data/src/datetime/timezones.ts [hunk 1]
```diff
@@ -148,13 +152,13 @@ const abbrevationWithoutOffset = (abbrevation: string): string => {
...
 const mapToInfo = (timeZone: TimeZone, timestamp: number): TimeZoneInfo | undefined => {
   const momentTz = moment.tz.zone(timeZone);
-
   if (!momentTz) {
     return undefined;
   }
 
   return {
     name: timeZone,
+    ianaName: momentTz.name,
```
Comment (mckn) [inline packages/grafana-data/src/datetime/timezones.ts:161]: I think the proper name is olsonId.

**Review by mckn — APPROVED** [review]

File: packages/grafana-data/src/datetime/timezones.ts (hunk 1 above)
Comment (dprokop) [reply packages/grafana-data/src/datetime/timezones.ts:161]: Did a quick check on this and seems like there is no clear winner;) I think I'll stay with iana as [Wikipedia states Olson database is occasionally used in as a name of this DB](https://en.wikipedia.org/wiki/Tz_database)

File: jest.config.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+process.env.TZ = 'UTC';
```
Comment (hugohaggmark) [inline jest.config.js:1]: I guess this is deliberate, but feels very strange.

**Review by hugohaggmark — APPROVED** [review]
Looks great Dominik, I just left a minor nit.

File: jest.config.js (hunk 2 above)
Comment (dprokop) [reply jest.config.js:1]: Yeah, had the same feeling but apparently this is the way jest wants the time zone to be set. I have not found any other way to get the timezone "mocked".

**Pull request merged**